core/store.go

```diff
@@ -290,6 +291,111 @@ func Get(k string) *Obj {
 	return getHelper(k, true)
 }
 
+var (
+	keyList []string
+)
+
+// initKeyList initializes keyList with keys from keypool if it is empty or outdated.
+// This function is used to populate keyList with all keys from keypool,
+// ensuring that it matches the current state of the keypool.
+func initKeyList() {
+	// Check if keyList is empty or needs to be updated
+	if len(keyList) == 0 || len(keyList) != len(keypool) {
+		// Initialize keyList with capacity equal to the length of keypool
+		keyList = make([]string, 0, len(keypool))
+		// Populate keyList with keys from keypool
+		for k := range keypool {
+			keyList = append(keyList, k)
+		}
+	}
+}
+
+// scanKeys performs a scan operation on the keypool, returning keys that match
+// the given criteria (cursor, count, pattern, and keyType).
+// Arguments:
+// - cursor: The starting point for the scan operation.
+// - count: The number of keys to return.
+// - pattern: A glob pattern to match against keys (can be empty).
+// - keyType: The type of keys to include in the result (can be empty).
+// Returns:
+// - newCursor: The new cursor position after the scan.
+// - keys: A slice of keys that match the criteria.
+func scanKeys(cursor, count int, pattern, keyType string) (newCursor int, keys []string) {
+	// Acquire locks for thread-safe access to keypool and store
+	withLocks(func() {
```

Comment:
The scan implementation in Redis seems to be a non-blocking operation. Wouldn't this block the DB operation?

Reply:
Yes this would block the operation with locks, need to think of better approach to make it non-blocking. If you could give some suggestions that would be of great help!